waypoint_generator/include/waypoint_generator/waypoint_generator.hpp

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) 2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#pragma once
+
+#include <ros/ros.h>
+#include <cav_msgs/Route.h>
+#include <autoware_msgs/LaneArray.h>
+#include <carma_wm/WMListener.h>
+#include <carma_wm/WorldModel.h>
+#include <vector>
+#include <tf/transform_datatypes.h>
+
+namespace waypoint_generator 
+{
+    /**
+     * The primary entrypoint class for the Waypoint Generator node. Handles 
+     * subscribing to the route and CARMA World Model and processing it into a
+     * form suitable form for converting to Autoware waypoints.
+     * 
+     * All indices of note (speed, orientations, centerline points) are assumed 
+     * to be according to the same index system, such that speed[i] and orientation[i]
+     * correspond to the overall (across all lanelets in the input route) i-th
+     * lanelet centerline point.
+     */
+    class WaypointGenerator 
+    {
+        public:
```

Comment:
I feel not all of its functions are used as its API. Does it make sense to have some helper function marked as private?

Reply:
I could think about that a bit, yeah. It's just a class that holds those methods so I can unit test them separately from the ROS API stuff. I can look into making some of those methods private/protected, if there's an easy way to unit test them still.